packages/frontend/src/components/VariablesList/index.tsx

```diff
@@ -0,0 +1,67 @@
+import type { ComponentType } from 'react'
+import List from '@mui/material/List'
+import ListItem, { ListItemProps } from '@mui/material/ListItem'
+import ListItemButton, {
+  ListItemButtonProps,
+} from '@mui/material/ListItemButton'
+import ListItemText from '@mui/material/ListItemText'
+import { Variable } from 'helpers/variables'
+
+function makeListItemComponent(
+  onClick?: (variable: any) => void,
+): ComponentType<ListItemButtonProps> | ComponentType<ListItemProps> {
+  if (onClick) {
+    return (props: ListItemButtonProps) => <ListItemButton {...props} />
+  }
+
+  return (props: ListItemProps) => <ListItem {...props} />
+}
+
+interface VariablesListProps {
+  variables: Variable[]
+  onClick?: (variable: any) => void
+  listHeight?: number
+}
+
+export default function VariablesList(props: VariablesListProps) {
+  const { variables, onClick, listHeight } = props
+  const ListItemComponent = makeListItemComponent(onClick)
+
+  if (!variables || variables.length === 0) {
+    return <></>
+  }
+
+  return (
+    <List
+      disablePadding
+      data-test="power-input-suggestion-group"
+      sx={{ maxHeight: listHeight, overflowY: 'auto' }}
+    >
+      {variables.map((variable) => (
+        <ListItemComponent
+          sx={{ pl: 4 }}
+          divider
+          data-test="power-input-suggestion-item"
+          key={`suggestion-${variable.name}`}
+          onClick={onClick && (() => onClick(variable))}
```

Comment:
nit: good catch. I'm generally not a big fan of passing in explicitly `undefined` props, but that's a stylistic choice. If you subscribe to this there's probably a cleaner way to refactor `makeListItemComponent` to avoid this need.